lib/chai/core/assertions.js

```diff
@@ -1285,130 +1286,104 @@ module.exports = function (chai, _) {
    * @name throw
    * @alias throws
    * @alias Throw
-   * @param {ErrorConstructor} constructor
-   * @param {String|RegExp} expected error message
+   * @param {Error|ErrorConstructor} errorLike
+   * @param {String|RegExp} errMatcher error message
    * @param {String} message _optional_
    * @see https://developer.mozilla.org/en/JavaScript/Reference/Global_Objects/Error#Error_types
    * @returns error for chaining (null if no error)
    * @namespace BDD
    * @api public
    */
 
-  function assertThrows (constructor, errMsg, msg) {
+  function assertThrows (errorLike, errMatcher, msg) {
     if (msg) flag(this, 'message', msg);
     var obj = flag(this, 'object');
+    var negate = flag(this, 'negate');
     new Assertion(obj, msg).is.a('function');
 
-    var thrown = false
-      , desiredError = null
-      , name = null
-      , thrownError = null;
-
-    if (arguments.length === 0) {
-      errMsg = null;
-      constructor = null;
-    } else if (constructor && (constructor instanceof RegExp || 'string' === typeof constructor)) {
-      errMsg = constructor;
-      constructor = null;
-    } else if (constructor && constructor instanceof Error) {
-      desiredError = constructor;
-      constructor = null;
-      errMsg = null;
-    } else if (typeof constructor === 'function') {
-      name = constructor.prototype.name;
-      if (!name || (name === 'Error' && constructor !== Error)) {
-        name = constructor.name || (new constructor()).name;
-      }
-    } else {
-      constructor = null;
+    if (errorLike instanceof RegExp || typeof errorLike === 'string' && arguments.length < 2) {
+      errMatcher = errorLike;
+      errorLike = null;
     }
```

Comment:
It's my understanding that this logic allows a user to overload the first parameter when they only care about error message and not type, allowing them to do `assertThrows("some message")` instead of `assertThrows(undefined, "some message")`.

But how does the `arguments.length < 2` play into the mix? And why is it only important if the first argument is a string, but not if the first argument is a regexp?

Just for giggles I tried removing it and running tests and nothing broke, so if there is an importance to it I recommend a comment be added to the code, and a test be added to verify functionality.